lib/commands/backup.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+const Command = require('../command');
+const Promise = require('bluebird');
+const createDebug = require('debug');
+const Zip = require('adm-zip');
+const fs = require('fs');
+const path = require('path');
+const cli = require('../');
+
+const debug = createDebug('ghost-cli:backup');
+
+function absolutePath(inputPath) {
+    return path.isAbsolute(inputPath) ? inputPath : path.join(process.cwd(), inputPath);
+}
+
+class BackupCommand extends Command {
+    run(argv) {
+        const instance = this.system.getInstance();
+        const datetime = (new Date()).toJSON().substring(0, 10);
+
+        this.database = {};
+        this.saveLocation = process.cwd();
+
+        if (argv.output) {
+            this.saveLocation = absolutePath(this.saveLocation);
+            fs.ensureDirSync(this.backupData.saveLocation);
+        }
+
+        this.saveLocation = path.join(this.saveLocation, `ghost.backup.${datetime}.zip`)
+
+        // The zip library used (adm-zip) has a known issue where the zip is structurally sound,
+        // but Windows has issues reading / unzipping it. To counteract this, all files are read
+        // and added via the `addFile` method, where the file permissions can be set, seemingly
+        // resolving the issue. See https://github.com/cthackers/adm-zip/issues/171
+        this.zipFile = new Zip();
+
+        if (instance.running()) {
+            this.ui.log('Ghost is currently running. Backing up might take longer and slow down your site', 'yellow');
+            instance.loadRunningEnvironment();
+        } else {
+            instance.checkEnvironment();
+        }
+
+        return this.ui.listr([{
+            title: 'Backing up plugins',
+            task: this.backupPlugins.bind(this)
+        }, {
+            title: 'Backing up database',
+            task: this.backupDatabase.bind(this)
+        }, {
+            title: 'Backing up config files',
+            task: this.backupConfig.bind(this)
+        }, {
+            title: 'Backing up content folder',
+            task: this.backupContent.bind(this)
+        }, {
+            title: `Writing backup file to ${this.saveLocation}`,
+            task: this.writeZipFile.bind(this)
+        }]);
+    }
+
+    backupContent() {
+        // Add content folder // @todo: Do we need to check if the content folder exists?
+        // @todo: Figure out how to deal with Windows bug
+        this.zipFile.addLocalFolder(path.join(process.cwd(), 'content/'), 'content/');
```

Comment:
So yeah, the windows bug isn't great. However, we should be able to fairly easily use another lib to recursively traverse the content folder and add it to the zip.

This library looks pretty good: https://github.com/jprichardson/node-klaw